test/libsolidity/syntaxTests/indexing/struct_array_noninteger_index.sol

```diff
@@ -0,0 +1,11 @@
+contract test {
+ struct s { uint a; uint b;}
+    function f() pure public returns (byte) {
+        s[75555555555555555555555555555555555555555555555555555555555555555555555555555555555555555555555555555555555555555555555555555555555555555555];
+        s[7];
+    }
+}
+
+// ----
+// TypeError: (101-241): Type int_const 7555...(132 digits omitted)...5555 is not implicitly convertible to expected type uint256.
+// TypeError: (101-241): Index expression cannot be represented as an unsigned integer.
```

Comment:
Actually this is a second error with exactly the same cause. We probably should not just create a new fatal error but handle it in a different way.

Reply:
If it *can* be handled differently without too many problems, then we can turn the fatal errors a few lines down in TypeChecker.cpp line 2210 into a non-fatal error as well.

Reply:
How about this?
```
	case Type::Category::TypeType:
	{
		TypeType const& typeType = dynamic_cast<TypeType const&>(*baseType);
		if (!index)
			resultType = make_shared<TypeType>(make_shared<ArrayType>(DataLocation::Memory, typeType.actualType()));
		else
		{
			u256 length = 0;
			if (expectType(*index, IntegerType::uint256()))
			{
				if (auto indexValue = dynamic_cast<RationalNumberType const*>(type(*index).get()))
					length = indexValue->literalValue(nullptr);
				else
					m_errorReporter.fatalTypeError(index->location(), "Integer constant expected.");
			}

			resultType = make_shared<TypeType>(make_shared<ArrayType>(
				DataLocation::Memory,
				typeType.actualType(),
				length
			));
		}
		break;
	}
```

Reply:
Please update the PR so we can properly see the diff.

Reply:
Maybe we could just proceed without any fatal error by using a RationalNumberType with value zero?

Reply:
Isn't that kind of what I did?

Reply:
Ah yeah - I hadn't noticed, because you still have the fatal error in the snippet - that can turn into a non-fatal one, if we fall back to just zero.

Reply:
Oh. Yes that is an other fatal error from the code before I changed anything. I figured my change should be as minimal-invasive as possible. I can remove that one too, I suppose.